### PR TITLE
samples: CAF Sensor Manager fix sysbuild regressions

### PR DESCRIPTION
**Pull request by KAGA164**
This fixes lack of proper adaptation to sysbuild on CAF Sensor Manager sample. It fix nRF54h20 and legacy platforms

### CODE REVIEW

**Review by peknis — APPROVED** [review]

**Review by grochu — APPROVED** [review]

**Comment by shanthanordic** [comment]
Dont merge without @ns-tolu  or @miha-nordic  approval

**Review by dchat-nordic — APPROVED** [review]

**Review by miha-nordic — APPROVED** [review]

**Comment by shanthanordic** [comment]
removed DNm has @miha-nordic  has approved.

**Pull request merged**
